instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpAttributesExtractor.java

```diff
@@ -43,7 +43,10 @@ protected final void onEnd(AttributesBuilder attributes, REQUEST request, RESPON
         attributes,
         SemanticAttributes.HTTP_REQUEST_CONTENT_LENGTH_UNCOMPRESSED,
         requestContentLengthUncompressed(request, response));
-    set(attributes, SemanticAttributes.HTTP_STATUS_CODE, statusCode(request, response));
+    Integer statusCode = statusCode(request, response);
+    if (statusCode != null) {
```

Comment:
I wonder if we should use `OptionalInt` instead?

Reply:
I think there's still reasonable chance of someone returning `null` even with `OptionalInt`, I guess nullable return has less chance of NPE.

Reply:
And we're never using any `Optional`s anywhere in the agent, so even if `OptionalInt` is a bit nicer and prettier ` @Nullable Integer` is more consistent with the codebase.